commit 03d07ee94c97814d4924602e9cefbf609c7e5ea0
Author: Dan Carpenter <dan.carpenter@oracle.com>
Date:   Fri Dec 21 12:11:20 2018 +0300

    ASoC: pcm512x: Fix a double unlock in pcm512x_digital_mute()
    
    [ Upstream commit 28b698b7342c7d5300cfe217cd77ff7d2a55e03d ]
    
    We accidentally call mutex_unlock(&pcm512x->mutex); twice in a row.
    
    I re-wrote the error handling to use "goto unlock;" instead of returning
    directly.  Hopefully, it makes the code a little simpler.
    
    Fixes: 3500f1c589e9 ("ASoC: pcm512x: Implement the digital_mute interface")
    Signed-off-by: Dan Carpenter <dan.carpenter@oracle.com>
    Reviwed-by: Dimitris Papavasiliou <dpapavas@gmail.com>
    Signed-off-by: Mark Brown <broonie@kernel.org>

diff --git a/sound/soc/codecs/pcm512x.c b/sound/soc/codecs/pcm512x.c
index 47cf3013ead5..e6523e74fdd9 100644
--- a/sound/soc/codecs/pcm512x.c
+++ b/sound/soc/codecs/pcm512x.c
@@ -1404,24 +1404,20 @@ static int pcm512x_digital_mute(struct snd_soc_dai *dai, int mute)
 		if (ret != 0) {
 			dev_err(component->dev,
 				"Failed to set digital mute: %d\n", ret);
-			mutex_unlock(&pcm512x->mutex);
-			return ret;
+			goto unlock;
 		}
 
 		regmap_read_poll_timeout(pcm512x->regmap,
 					 PCM512x_ANALOG_MUTE_DET,
 					 mute_det, (mute_det & 0x3) == 0,
 					 200, 10000);
-
-		mutex_unlock(&pcm512x->mutex);
 	} else {
 		pcm512x->mute &= ~0x1;
 		ret = pcm512x_update_mute(pcm512x);
 		if (ret != 0) {
 			dev_err(component->dev,
 				"Failed to update digital mute: %d\n", ret);
-			mutex_unlock(&pcm512x->mutex);
-			return ret;
+			goto unlock;
 		}
 
 		regmap_read_poll_timeout(pcm512x->regmap,
@@ -1432,9 +1428,10 @@ static int pcm512x_digital_mute(struct snd_soc_dai *dai, int mute)
 					 200, 10000);
 	}
 
+unlock:
 	mutex_unlock(&pcm512x->mutex);
 
-	return 0;
+	return ret;
 }
 
 static int pcm512x_set_tdm_slot(struct snd_soc_dai *dai,
diff -uN a/1.txt b/1.txt
--- a/dummy/rpi_1811_03d07ee94c97814d4924602e9cefbf609c7e5ea0.txt	1970-01-01 00:00:00.000000000 +0000
+++ b/dummy/rpi_1811_03d07ee94c97814d4924602e9cefbf609c7e5ea0.txt	2013-12-23 04:07:40.000000000 +0000
@@ -0,0 +1 @@
+dummy file to ensure patch has content.
